stock_no_negative/models/stock_quant.py

```diff
@@ -22,7 +22,7 @@ def check_negative_qty(self):
         )
         if not check_negative_qty:
             return
-
+        self._merge_quants()
```

Comment:
Add comment for knowing why is here in later references
```suggestion
        # Execute generic merge method before doing comparisons for avoiding
        # existing negative quants not merged with their counterparts
        self._merge_quants()
```